### DISCUSSION TITLE
Separate thread group or single for performance test with iso8583?

### DISCUSSION THREAD

**Discussion by mehrdad2000** [tilln/jmeter-iso8583#92] [General]
Hi
For performance test which is better using separate thread group or single? Why?

e.g. i have 3 type of message that create a sampler for each one, now question is should I separate thread group or put all of them in single threads group?

What is the relation of this with the option “same user on each iteration”?

any idea?
Thanks

**Comment by tilln** [comment]
> better using separate thread group or single? Why?

Depends on your use case.

> 3 type of message that create a sampler for each one

Both work, but both have some downside.
- Single Thread Group: You need additional script logic to select the type, e.g. Switch Controller, If Controller.
- Separate Thread Groups: You need to control the throughput of each individually.

> option “same user on each iteration”

Relevant for HTTP sampler caching and cookies, not relevant for ISO8583.

Sounds like you may want to acquire some basic JMeter skills first, though this is not the right forum, sorry.

Try Stackoverflow or the JMeter groups?
